script.js

```diff
@@ -0,0 +1,165 @@
+const toDoForm = document.querySelector("form");
+const toDoInput = toDoForm.querySelector(".input-text");
+const toDoButton = toDoForm.querySelector(".input-button");
+
+const TODO_DB = "todo";
+let toDoList = [];
+
+// localStorage 모델: 웹이 로드 될 때 기존 투두리스트를 가져오기 위함
+const localStorageModel = {
+    // localStorage에 저장
+    save: () => {
+        localStorage.setItem(TODO_DB, JSON.stringify(toDoList));
+    },
+
+    // localStorage에서 로드
+    load: () => {
+        const loaded = localStorage.getItem(TODO_DB);
+        if (loaded !== null) {
+            const parsed = JSON.parse(loaded);
+            toDoList = parsed;
+        }
+    }
+}
+
+// List 모델
+const list = {
+    // 개수 계산
+    getCount: () => {
+        let pending = 0;
+        toDoList.forEach((item) => {
+            if (!item.done) {
+                pending++;
+            }
+        })
+        return {
+            pending: pending,
+            done: toDoList.length - pending
+        }
+    },
+
+    // list 대기중 완료 토글
+    updateList: (value, isDone) => {
+        const i = toDoList.findIndex((item) => {
+            return item.text == value;
+        })
+        toDoList[i].done = toDoList[i].done == isDone ? !isDone : isDone;
+    },
+
+    // toDoList 배열에 추가  -> submit 할 때
+    addToList: (value) => {
+        let item = {
+            "id": toDoList.length + 1,
+            "text": value,
+            "done": false
```

Comment:
속성은 string으로 지정할 필요는 없습니다! 이것과 완전히 동일한 코드입니다.
```suggestion
            id: toDoList.length + 1,
            text: value,
            done: false
```

그리고, todo객체가 어떤 schema(형태)를 가져야 하는지 주석으로 간단하게라도 적어 놓으면 좋을 것 같네요! todo 객체가 어떻게 생겼는지를 모르니까요 ㅠㅠ. 나중에 TypeScript를 이용하면 또 이렇게 아예 타입을 선언해버릴 수도 있습니다.

```typescript
interface Todo {
    text: string;
    done: boolean;
}
```

Reply:
오 좋은 생각이에요 todo 객체에 들어가는 내용을 선언할 때 적어 놓으면 나중에 코드를 다시 볼 때 해석하기 편하기도 하겠네요!! TypeScript는 아직 어떤 것인지 모르지만, 저렇게 타입을 선언하는 방식은 Swift언어와 비슷하네요 ㅎㅎ